samples/boards/nrf/system_off/src/retained_mem.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "retained.h"
+
+#include <stdint.h>
+#include <string.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/retained_mem.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/crc.h>
+
+const static struct device *retained_mem_device = DEVICE_DT_GET(DT_ALIAS(retainedmemdevice));
+
+struct retained_data retained;
+
+#define RETAINED_CRC_OFFSET offsetof(struct retained_data, crc)
+#define RETAINED_CHECKED_SIZE (RETAINED_CRC_OFFSET + sizeof(retained.crc))
+
+bool retained_validate(void)
```

Comment:
this function looks very like https://github.com/zephyrproject-rtos/zephyr/blob/34e4edc28a3806885821d863cc3ebc9c60f5b817/samples/boards/nrf/system_off/src/retained.c#L139 Why not merge this into a single file and avoid redundancy?

Reply:
I agree, the difference between those two is so small they should be merged.

Reply:
fixed